### PR TITLE
feat(Rich Text Editor): list plugin with soft break

### PR DESCRIPTION
**Pull request by bojangles-m**
Added option for soft break to behave exactly as break.

### CODE REVIEW

File: src/components/RichTextEditor/Plugins/ListPlugin/ListPlugin.ts [hunk 1]
```diff
@@ -15,15 +17,28 @@ export const createLicStylePlugin = createPluginFactory<ToggleMarkPlugin>({
...
+        this.isSoftBreak = props?.isSoftBreak ?? false;
     }
 
     plugins() {
-        return [createListPlugin(), createLicStylePlugin()];
+        return [
+            createListPlugin({
+                withOverrides: withList,
+                options: {
+                    isSoftBreak: this.isSoftBreak,
```
Comment (triggertoo) [inline src/components/RichTextEditor/Plugins/ListPlugin/ListPlugin.ts:38]: maybe rename to `useSoftBreak`?

**Review by triggertoo — COMMENTED** [review]
looks good 👍 Could you add a test for the behaviour?

File: src/components/RichTextEditor/Plugins/ListPlugin/ListPlugin.ts (hunk 1 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/Plugins/ListPlugin/ListPlugin.ts:38]: Would you have some other name for it, this is more like Hook name :)

**Comment by bojangles-m** [comment]
> looks good 👍 Could you add a test for the behaviour?

Do you have an idea how this test would look like?

File: src/components/RichTextEditor/Plugins/ListPlugin/ListPlugin.ts (hunk 1 above)
Comment (triggertoo) [reply src/components/RichTextEditor/Plugins/ListPlugin/ListPlugin.ts:38]: imho it's unrelated to react, but we can also keep it that way.

**Comment by triggertoo** [comment]
> > looks good 👍 Could you add a test for the behaviour?
> 
> Do you have an idea how this test would look like?

you could check if a softbreak (shift-enter) in a list creates a new list item

File: src/components/RichTextEditor/__tests__/ListPlugin.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        cy.get(TEXTSTYLE_DROPDOWN_TRIGGER).click({ force: true });
+        cy.get(TEXTSTYLE_OPTION).first().click();
+        cy.get(firstListItemSelector).should(
+            'have.attr',
+            'style',
+            'font-size: 48px; font-style: normal; font-weight: 700;',
+        );
+    });
+
+    it.skip('renders the list item without text decoration underline', () => {
```
Comment (triggertoo) [inline src/components/RichTextEditor/__tests__/ListPlugin.cy.tsx:56]: please remove skip to execute the test

File: src/components/RichTextEditor/__tests__/ListPlugin.cy.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        cy.mount(<RichTextEditorWithUnorderedListStyles />);
+
+        cy.get('[contenteditable=true] li').should('have.class', '!tw-no-underline');
+    });
+
+    it.skip('renders custom styled list items', () => {
```
Comment (triggertoo) [inline src/components/RichTextEditor/__tests__/ListPlugin.cy.tsx:62]: please remove skip to execute the test

File: src/components/RichTextEditor/__tests__/ListPlugin.cy.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    it('renders new list item with soft break', () => {
+        cy.mount(<RichTextEditorWithUnorderedListStyles isSoftBreak={true} />);
+        const itemValue = 'New item in list';
+
+        const firstListItemSelector = '[contenteditable=true] ul:first-child > li:first-child';
+        const secondListItemSelector = '[contenteditable=true] ul:first-child > li:nth-child(2)';
+
+        cy.get(firstListItemSelector).type(`{end}{shift+enter}${itemValue}`);
+
+        cy.get(firstListItemSelector).children().should('have.length', 1);
```
Comment (triggertoo) [inline src/components/RichTextEditor/__tests__/ListPlugin.cy.tsx:85]: you could check if the number of list items is equal 2 to make sure, a list item was added

File: src/components/RichTextEditor/__tests__/ListPlugin.cy.tsx (hunk 2 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/__tests__/ListPlugin.cy.tsx]: of course, thanx, forget when i suppressed the test

File: src/components/RichTextEditor/__tests__/ListPlugin.cy.tsx (hunk 3 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/__tests__/ListPlugin.cy.tsx]: udpated

File: src/components/RichTextEditor/__tests__/ListPlugin.cy.tsx (hunk 4 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/__tests__/ListPlugin.cy.tsx:87]: udpated

Comment (triggertoo) [reply src/components/RichTextEditor/__tests__/ListPlugin.cy.tsx:87]: it seems a bit confusing if the length is 4 here. what if you instead just initialise the editor with just one children li element (not loading `unorderedListValue` from the helper but defining it yourself. it's easier to understand if the test fails.

**Review by triggertoo — APPROVED** [review]

**Pull request merged**